cfme/tests/test_db_restore.py

```diff
@@ -1,131 +1,89 @@
-# -*- coding: utf-8 -*-
-
 import fauxfactory
 import pytest
 
-from cfme.cloud import provider as cloud_provider
 from cfme.cloud.provider.ec2 import EC2Provider
 from cfme.common.vm import VM
-from cfme.infrastructure.provider import wait_for_a_provider
 from cfme.infrastructure.provider.virtualcenter import VMwareProvider
 from fixtures.pytest_store import store
-from cfme.utils.appliance import provision_appliance
 from cfme.utils.log import logger
 from cfme.utils.providers import list_providers_by_class
-from cfme.utils import version
-
-
-@pytest.fixture
-def virtualcenter_provider_crud():
-    try:
-        return list_providers_by_class(VMwareProvider)[0]
-    except IndexError:
-        pytest.skip("No VMware providers available (required)")
 
 
-@pytest.fixture
-def ec2_provider_crud():
+def provider_app_crud(provider_class, appliance):
     try:
-        return list_providers_by_class(EC2Provider)[0]
+        prov = list_providers_by_class(provider_class)[0]
+        prov.appliance = appliance
+        return prov
     except IndexError:
-        pytest.skip("No EC2 provider available (required)")
+        pytest.skip("No {} providers available (required)".format(provider_class.type))
 
 
 def provision_vm(request, provider):
     """Function to provision appliance to the provider being tested"""
     vm_name = "test_rest_db_" + fauxfactory.gen_alphanumeric()
     vm = VM.factory(vm_name, provider)
-
     request.addfinalizer(vm.delete_from_provider)
-
     if not provider.mgmt.does_vm_exist(vm_name):
         logger.info("deploying %s on provider %s", vm_name, provider.key)
         vm.create_on_provider(allow_skip="default")
     else:
         logger.info("recycling deployed vm %s on provider %s", vm_name, provider.key)
     vm.provider.refresh_provider_relationships()
-    vm.wait_to_appear()
     return vm
 
 
 @pytest.fixture(scope="module")
-def get_appliances():
+def get_appliances(temp_appliances_unconfig_modscope):
     """Returns two database-owning appliances
 
     """
-    ver_to_prov = str(version.current_version())
-    appl1 = provision_appliance(ver_to_prov, 'test_back')
-    appl2 = provision_appliance(ver_to_prov, 'test_rest')
+    appl1 = temp_appliances_unconfig_modscope[0]
+    appl2 = temp_appliances_unconfig_modscope[1]
     appl1.configure(region=0)
-    appl1.ipapp.wait_for_web_ui()
+    appl1.wait_for_web_ui()
     appl2.configure(region=0)
-    appl2.ipapp.wait_for_web_ui()
-    return (appl1, appl2)
+    appl2.wait_for_web_ui()
+    return temp_appliances_unconfig_modscope
 
 
+# TODO Refactore test in to fixtures
 @pytest.mark.tier(2)
 @pytest.mark.uncollectif(
     lambda: not store.current_appliance.is_downstream)
-def test_db_restore(request, soft_assert, virtualcenter_provider_crud, ec2_provider_crud,
-                    appliance):
-
-    appl1, appl2 = get_appliances()
-
-    def finalize():
-        appl1.destroy()
-        appl2.destroy()
-    request.addfinalizer(finalize)
```

Comment:
**QUESTION**
Why did you remove the finalizer?

Reply:
This is handled by the appliance fixture now I thought? https://github.com/ManageIQ/integration_tests/blob/master/fixtures/appliance.py#L41